### PR TITLE
Fix/add missing tests

### PR DESCRIPTION
**Pull request by mrsasha**
@skraynick here it is!

### CODE REVIEW

File: android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    //use to get a reference?
+
+    @Rule
+    public final ServiceTestRule mServiceRule = new ServiceTestRule();
+
+    private SensorbergSdk tested;
+
+    @Before
+    public void setup() {
+        // this is one way of solving the "Can't create handler inside thread that has not called Looper.prepare()" error when creating
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java]: maybe make a proper /* */ multi line comment.

File: android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            public void presentBeaconEvent(BeaconEvent beaconEvent) {
+                //do nothing;
+            }
+        };
+
+        tested.registerEventListener(listener);
+        tested.hostApplicationInForeground();
+
+        Assertions.assertThat(SensorbergSdk.listeners.size()).isEqualTo(1);
+        verify(tested, times(2)).registerForPresentationDelegation(); //called once by registerEventListener and once by hostApplicationInForeground
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java]: comment above.

File: android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    @Test
+    public void hostApplicationInForeground_without_listeners_doesnt_register_messenger_with_service() {
+        tested.hostApplicationInForeground();
+
+        Assertions.assertThat(SensorbergSdk.listeners.size()).isEqualTo(0);
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java]: no line needed.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceIntentHandlingTests.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    @Test
+    public void should_handle_intent_with_start_message_and_api_key() {
+        Intent sensorbergServiceStartIntent = SensorbergServiceIntents
+                .getStartServiceIntent(InstrumentationRegistry.getContext(), TestConstants.API_TOKEN_DEFAULT);
+
+        int handleIntentResult = tested.handleIntent(sensorbergServiceStartIntent);
+
+        Assertions.assertThat(handleIntentResult).isEqualTo(SensorbergService.START_STICKY);
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceIntentHandlingTests.java]: no line needed.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+        verify(fileManager, times(2)).write(any(SensorbergServiceConfiguration.class), any(String.class));
+    }
+
+    @Test
+    public void should_turn_debugging_on_in_transport_from_intent() {
+        Intent serviceDebuggingOnIntent = SensorbergServiceIntents.getServiceLoggingIntent(InstrumentationRegistry.getContext(), true);
+
+        tested.handleDebuggingIntent(serviceDebuggingOnIntent, InstrumentationRegistry.getContext(), false);
+
+//        try {
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java]: if we don't need it in here, let's take out the commented lines.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+        Assertions.assertThat(Logger.log).isInstanceOf(Logger.VerboseLogger.class);
+    }
+
+    @Test
+    public void should_turn_debugging_off_in_transport_from_intent() {
+        Intent serviceDebuggingOffIntent = SensorbergServiceIntents.getServiceLoggingIntent(InstrumentationRegistry.getContext(), false);
+
+        tested.handleDebuggingIntent(serviceDebuggingOffIntent, InstrumentationRegistry.getContext(), false);
+
+//        try {
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java]: As above.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+    @Test
+    public void registerPresentationDelegate_should_add_delegate_to_messengerList() {
+        Messenger messenger = new Messenger(new Handler(InstrumentationRegistry.getContext().getMainLooper()));
+        tested.presentationDelegates = spy(tested.presentationDelegates);
+
+        try {
+            tested.registerPresentationDelegate(SensorbergServiceIntents.getIntentWithReplyToMessenger(InstrumentationRegistry.getContext(),
+                    SensorbergServiceMessage.MSG_REGISTER_PRESENTATION_DELEGATE, messenger));
+        } catch (Exception e) {
+            //this happens because the MessengerList, as an internal class of SensorbergService, has its own instance which is different
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java]: /* */

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+            //from the proxied spy instance we use for testing; MessengerList.add() calls bootstrapper.sentPresentationDelegationTo()
+            //and that will cause the NPE
+            Assertions.assertThat(e).isInstanceOf(NullPointerException.class);
+        }
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java]: take out excess lines

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+    @Test
+    public void unregisterPresentationDelegate_should_remove_delegate_from_messengerList() {
+        Messenger messenger = new Messenger(new Handler(InstrumentationRegistry.getContext().getMainLooper()));
+        tested.presentationDelegates = spy(tested.presentationDelegates);
+
+        try {
+            tested.unregisterPresentationDelegate(SensorbergServiceIntents.getIntentWithReplyToMessenger(InstrumentationRegistry.getContext(),
+                    SensorbergServiceMessage.MSG_UNREGISTER_PRESENTATION_DELEGATE, messenger));
+        } catch (Exception e) {
+            //this happens because the MessengerList, as an internal class of SensorbergService, has its own instance which is different
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceInternalTests.java]: /* */

**Comment by skraynick** [comment]
@mrsasha a few changes needed. I will look again quickly tomorrow morning.

File: android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java (hunk 3 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/SensorbergSdkTests.java]: I always put a line between execution and verification/assertion calls, that way the difference is more visible.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceIntentHandlingTests.java (hunk 4 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/SensorbergServiceIntentHandlingTests.java]: like above.

**Pull request merged**